plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/ChatController.kt

```diff
@@ -103,6 +109,28 @@ class ChatController private constructor(
         authController = AuthController(),
     )
 
+    init {
+        // Automatically start the project context LSP after some delay when average CPU load is below 30%.
+        // The CPU load requirement is to avoid competing with native JetBrains indexing and other CPU expensive OS processes
+        if (CodeWhispererSettings.getInstance().isProjectContextEnabled()) {
+            val scope = projectCoroutineScope(context.project)
+            scope.launch {
+                delay(60_000) // Wait for 60 seconds to get accurate CPU load for 1 min
```

Comment:
`yieldAndWaitForDumbModeEnd` would be better

Reply:
actually use `waitForSmartMode` because the former looks like it forces a modal popup

Reply:
Cannot call `waitForSmartMode` after `ProjectContextController.getInstance(project = context.project)`. I don't find a usage of `waitForSmartMode`

Reply:
Can we start this with a minimum of cpu core running? And then when users start `@workspace` querying we will start respect the # of cpu setup? (Is it possible to change # of cpus in the middle of indexing?)

Suggesting the above since I would usually look at my resource monitor when my IDE is slow, and I don't want to see the top 2 cpu usage are Intellij + Amazon Q Helper which takes an additional 200% for 10 mins.

Reply:
```kt
import com.intellij.openapi.project.waitForSmartMode
...
context.project.waitForSmartMode()
```

Reply:
>(Is it possible to change # of cpus in the middle of indexing?)
This is not doable.

> Intellij
This native indexing can take 800% (66% of my total CPU)